includes/forms.inc

```diff
@@ -515,3 +537,20 @@ function islandora_ip_embargo_object_embargo_form_submit($form, &$form_state) {
   }
   return $form;
 }
+
+/**
+ * Validate hexidecimal color input.
+ *
+ * @param array $element
+ *   The Drupal element data.
+ * @param array $form_state
+ *   The Drupal form state.
+ * @param array $form
+ *   The Drupal form definition.
+ */
+function islandora_ip_embargo_hex_validate($element, &$form_state, $form) {
+  $color = $element['#value'];
+  if (!ctype_xdigit($color) && strlen($color) === 6) {
```

Comment:
The validation only works if the length is 6, I think someone could use shorthand hex here? The regular expression used by color_valid_hexadecimal_string() accounts for the shorthand case, or I guess this implementation could check if length = 3 or 6.

Not a big deal, but I think if the validation function is here, it may as well work as a user may expect.